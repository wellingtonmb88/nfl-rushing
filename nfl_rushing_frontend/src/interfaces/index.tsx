import React from 'react';

export type Order = 'asc' | 'desc';

export interface EnhancedTableHeadProps {
  numSelected: number;
  onRequestSort: (event: React.MouseEvent<unknown>, property: keyof Player) => void;
  onSelectAllClick: (event: React.ChangeEvent<HTMLInputElement>, checked: boolean) => void;
  order: Order;
  orderBy: string;
  rowCount: number;
  rowsPerPage: number;
}

export interface Player {
  name: string;
  team: string;
  position: string;
  longestRush: string;
  rushingAttempts: number;
  rushingAttemptsPerGame: number;
  totalRushingYards: number;
  rushingAvgYardsPerAttempt: number;
  rushingYardsPerGame: number;
  totalRushingTouchdowns: number;
  rushingFirstDowns: number;
  rushingFirstDownsPercentage: number;
  rushingTwentyPlusYeards: number;
  rushingFortyPlusYeards: number;
  rushingFumbles: number;
}

export interface HeadCell {
  disablePadding: boolean;
  id: keyof Player;
  label: string;
  numeric: boolean;
}

export interface EnhancedTableToolbarProps {
  numSelected: number;
}

export interface EnhancedTableProps {
  currentSearchName: string;
}

export interface TableRowsProps {
  row: Player;
  isItemSelected: boolean;
  labelId: string;
  handleClick: (player: Player) => void;
}
 
