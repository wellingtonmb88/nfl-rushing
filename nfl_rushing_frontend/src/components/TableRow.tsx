import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Checkbox from '@material-ui/core/Checkbox';

import { TableRowsProps } from '../interfaces'

export default (props: TableRowsProps) => {
  const { row, handleClick, isItemSelected, labelId } = props; 
  return (
    <TableRow
      hover
      onClick={event => handleClick(row)}
      role="checkbox"
      aria-checked={isItemSelected}
      tabIndex={-1}
      key={row.name}
      selected={isItemSelected}
    >
      <TableCell padding="checkbox">
        <Checkbox
          checked={isItemSelected}
          inputProps={{ 'aria-labelledby': labelId }}
        />
      </TableCell>
      <TableCell component="th" id={labelId} scope="row" padding="none">
        {row.name}
      </TableCell>
      <TableCell align="left">{row.team}</TableCell>
      <TableCell align="left">{row.position}</TableCell>
      <TableCell align="right">{row.longestRush}</TableCell>
      <TableCell align="right">{row.rushingAttempts}</TableCell>
      <TableCell align="right">{row.rushingAttemptsPerGame}</TableCell>
      <TableCell align="right">{row.totalRushingYards}</TableCell>
      <TableCell align="right">{row.rushingAvgYardsPerAttempt}</TableCell>
      <TableCell align="right">{row.rushingYardsPerGame}</TableCell>
      <TableCell align="right">{row.totalRushingTouchdowns}</TableCell>
      <TableCell align="right">{row.rushingFirstDowns}</TableCell>
      <TableCell align="right">{row.rushingFirstDownsPercentage}</TableCell>
      <TableCell align="right">{row.rushingTwentyPlusYeards}</TableCell>
      <TableCell align="right">{row.rushingFortyPlusYeards}</TableCell>
      <TableCell align="right">{row.rushingFumbles}</TableCell>
    </TableRow>
  );
}