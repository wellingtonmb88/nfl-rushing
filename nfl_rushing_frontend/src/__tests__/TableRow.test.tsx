import React from "react";
import { create } from "react-test-renderer";
import TableRow from '../components/TableRow'
import { Player } from '../interfaces'

const player: Player = {
  "longestRush": "11",
  "name": "Adam Thielen",
  "position": "WR",
  "rushingAttempts": 2,
  "rushingAttemptsPerGame": 0.1,
  "rushingAvgYardsPerAttempt": 7.5,
  "rushingFirstDowns": 1,
  "rushingFirstDownsPercentage": 50,
  "rushingFortyPlusYeards": 0,
  "rushingFumbles": 0,
  "rushingTwentyPlusYeards": 0,
  "rushingYardsPerGame": 0.9,
  "team": "MIN",
  "totalRushingTouchdowns": 0,
  "totalRushingYards": 15
}

describe("TableRow component", () => {
  test("TableRow with Snapshot", () => {
    const component = create(<TableRow
      row={player}
      isItemSelected={false}
      labelId='labelId-TableRow'
      handleClick={() => { }}
    />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});