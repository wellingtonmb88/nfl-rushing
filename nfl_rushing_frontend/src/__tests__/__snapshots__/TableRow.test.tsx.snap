// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow component TableRow with Snapshot 1`] = `
<tr
  aria-checked={false}
  className="MuiTableRow-root MuiTableRow-hover"
  onClick={[Function]}
  role="checkbox"
  tabIndex={-1}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-paddingCheckbox"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-24 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          aria-labelledby="labelId-TableRow"
          checked={false}
          className="PrivateSwitchBase-input-27"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
    </span>
  </td>
  <th
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-paddingNone"
    id="labelId-TableRow"
    scope="row"
  >
    Adam Thielen
  </th>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignLeft"
  >
    MIN
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignLeft"
  >
    WR
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    11
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    2
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    0.1
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    15
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    7.5
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    0.9
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    0
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    1
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    50
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    0
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    0
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    0
  </td>
</tr>
`;
